primitives/src/swaps.rs

```diff
@@ -1,15 +1,40 @@
-use crate::types::Asset;
+use crate::types::{Asset, OutcomeReport, PoolStatus};
 use alloc::vec::Vec;
-use frame_support::dispatch::DispatchError;
+use frame_support::dispatch::{DispatchError, DispatchResult};
 
 pub trait Swaps<AccountId> {
     type Balance;
     type MarketId;
 
+    /// Creates an initial active pool.
+    ///
+    /// # Arguments
+    ///
+    /// * `who`: The account that is the creator of the pool. Must have enough
+    /// funds for each of the assets to cover the `MinLiqudity`.
+    /// * `assets`: The assets that are used in the pool.
+    /// * `swap_fee`: The fee applied to each swap.
+    /// * `weights`: These are the denormalized weights (the raw weights).
     fn create_pool(
         creator: AccountId,
         assets: Vec<Asset<Self::MarketId>>,
         swap_fee: Self::Balance,
         weights: Vec<u128>,
     ) -> Result<u128, DispatchError>;
+
+    /// Returns the assets of a pool.
+    fn pool_assets(pool_id: u128) -> Result<Vec<Asset<Self::MarketId>>, DispatchError>;
```

Comment:
Maybe we can create a type alias `PoolId` for `u128` like we have for market `MarketId`. Not sure if we ever change this, but u64 is likely sufficient and most 64bit processors have instructions to directly manipulate them in one register, therefore being more efficient in handling them.

Reply:
> Maybe we can create a type alias PoolId for u128 like we have for market MarketId

Done

> Not sure if we ever change this, but u64 is likely sufficient and most 64bit processors have instructions to directly manipulate them in one register, therefore being more efficient in handling them.

For runtime, `wasm32-unknown-unknown` has a pointer size of 32 bits and the change to `u64` should probably create more optimized assembly, although I am not sure about its feasibility.